File: dotnet/ExtentReports/Source/CategoryHtml.cs
// ***********************************************************************
// Copyright (c) 2015, Anshoo Arora (Relevant Codes). All rights reserved.
//
// Copyrights licensed under the New BSD License.
//
// See the accompanying LICENSE file for terms.
// ***********************************************************************

namespace RelevantCodes.ExtentReports.Source
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class CategoryHtml
    {
        public static string GetOptionSource()
        {
            return "<option value='<!--%%TESTCATEGORYU%%-->'><!--%%TESTCATEGORY%%--></option>";
        }

        public static string GetCategoryViewSource()
        {
            return "<div class='col s12 m12 l12'>" +
                    "<div class='card-panel category-view'>" +
                        "<div class='category-header test-attributes'>"
                            + "<span class='category'><!--%%CATEGORYVIEWNAME%%--></span>"
                            + "<div class='category-status right'>"
                                + "<span class='cat-pass'>PASS: </span>"
                                + "<span class='cat-fail'>FAIL: </span>"
                                + "<span class='cat-other'>OTHER: </span>"
                            + "</div>"
                        + "</div>" +
                        "<table class='bordered'>" +
                            "<tr>" +
                                "<th>Run Date</th>" +
                                "<th>Test Name</th>" +
                                "<th>Status</th>" +
                            "</tr>" +
                            "<!--%%CATEGORYVIEWTESTDETAILS%%-->" +
                        "</table>" +
                    "</div>" +
                "</div>";
        }

        public static string GetCategoryViewTestSource()
        {
            return "<tr>" +
                    "<td><!--%%CATEGORYVIEWTESTRUNTIME%%--></td>" +
                    "<td><span class='category-link'><!--%%CATEGORYVIEWTESTNAME%%--></span></td>" +
                    "<td><div class='label <!--%%CATEGORYVIEWTESTSTATUS%%-->'><!--%%CATEGORYVIEWTESTSTATUS%%--></div></td>" +
                "</tr>";
        }
    }
}

File: java/extentreports/src/main/java/com/relevantcodes/extentreports/source/CategoryHtml.java
/*
* Copyright (c) 2015, Anshoo Arora (Relevant Codes).  All rights reserved.
* 
* Copyrights licensed under the New BSD License.
* 
* See the accompanying LICENSE file for terms.
*/

package com.relevantcodes.extentreports.source;

public class CategoryHtml {
    public static String getOptionSource() {
        return "<option class='cat-option'></option>";        
    }
    
    public static String getCategoryViewSource() {
		return "<li class='category-item displayed'>" +
				"<div class='cat-head'>" +
					"<span class='category-name'></span>" +
				"</div>" +
				"<div class='category-status-counts'>" +
					"<span class='cat-pass label'></span>" +
					"<span class='cat-fail label'></span>" +
					"<span class='cat-other label'></span>" +
				"</div>" +
				"<div class='cat-body'>" +
					"<div class='category-status-counts'>" +
						"<span class='cat-pass label'></span>" +
						"<span class='cat-fail label'></span>" +
						"<span class='cat-other label'></span>" +
					"</div>" +
					"<div class='cat-tests'>" +
						"<table class='bordered'>" +
							"<thead>" +
								"<tr>" +
									"<th>Run Date</th>" +
									"<th>Test Name</th>" +
									"<th>Status</th>" +
								"</tr>" +
							"</thead>" +
						"</table>" +
					"</div>" +
				"</div>"+ 
			"</li>";
    }
    
    public static String getCategoryViewTestSource() {
        return "<table><tr>" +
                "<td><!--%%CATEGORYVIEWTESTRUNTIME%%--></td>" +
                "<td><span class='category-link linked'><!--%%CATEGORYVIEWTESTNAME%%--></span></td>" +
                "<td><div class='status label capitalize '><!--%%CATEGORYVIEWTESTSTATUS%%--></div></td>" +
            "</tr></table>";
    }
}
